### ISSUE TITLE
Inconsistent base64 encoding during kubernetes deployment

### ISSUE THREAD

**Issue by asnare** [schipholgroup/takeoff#59]
During `deploy_to_kubernetes`, the `deployment.yml` configuration is rendered as a Jinja template. The variables that are available for substitution come from several places, including custom variables and Azure Keyvaults. The documentation currently states that variables from Azure Keyvaults are base64-encoded before substitution while variables from other sources are not.

The documentation, however, seems to be incorrect. Currently the behaviour seems to be that irrespective of source takeoff guesses whether the variable is already base64-encoded or not. If not, it gets encoded. If it is, it is left alone. This guessing is often wrong: it's essentially a heuristic and there are lots of non-encoded values that appear to be base64-encoded. (For example, `the quick brown fox jumps over the lazy dog`.)

Aside from the documentation mismatch, the impact of this is that variable interpolation is unpredictable and difficult to diagnose when issues go wrong. There is a workaround available: always base64-encode values in the Keyvault. (These will be left alone by takeoff and can be used as-is in the `data` section of a Secret, or decoded with a Jinja filter if the original value is required.)

Stepping back a bit, the purpose of this feature is to help with ensure that secrets stay so: values from the key vault should only be used in the `Secret` resource, which expects base64-encoded in the `data` attribute. This is laudable, but the current mechanism is error-prone.

Some ideas for resolving this issue:

 - Update the implementation to match the documentation: values from the vault are unconditionally base64-encoded, other values are left alone.
 - Leave all values alone. The user can then control base64 encoding (or not) using Jinja filters, which makes it easier to see what's going on and less magical. The user knows the context, and can decide appropriately what should be done.

From my perspective, I'd prefer the second option because it's more explicit. However it's most important that it behave predictably.

**Comment by danielvdende** [comment]
I'm inclined towards option 2 as well. To be honest, when we implemented this, I don't think we were aware of the `stringData` field in Kubernetes secrets. That essentially performs this logic. I'm in favour of deferring this kind of logic to underlying deployment targets (Kubernetes in this case).

**Comment by bjuli** [comment]
I agree with option 2 as well. I would like to also pick this up and open an PR soon.
